lite/provider.go

```diff
@@ -30,3 +33,105 @@ type PersistentProvider interface {
 	// SaveFullCommit saves a FullCommit (without verification).
 	SaveFullCommit(fc FullCommit) error
 }
+
+// A provider that can update itself w/ more recent commit data.
+type UpdatingProvider interface {
```

Comment:
How is the Updating provider different from the persistent provider?

Reply:
The persisitent provider has the possibility to store trusted fullcommits via `SaveFullCommit`. You guys use this in `UpdatingProvider` in `fillValsetAndSaveFC` to store the trusted commit.